### DISCUSSION TITLE
Seeking help: `c c` opening `COMMIT_EDITMSG` in a new frame

### DISCUSSION THREAD

**Discussion by nrnrnr** [magit/magit#4443] [Support] [answered]
When I go to commit changes, Magit is opening `COMMIT_EDITMSG` in a new frame, rather than a new window.  I'd love some help resolving this issue.

I'm using these versions, as reported by `magit-version`:

    Magit 3.1.1, Git 2.20.1, Emacs 27.1, gnu/linux

Here's what's wrong:

  - On typing `c c` in a Magit window, I expected Magit to open `COMMIT_EDITMSG` in the other window.

  - Instead, Magit opens a new frame. The frame is split into two
    windows. The upper window shows `COMMIT_EDITMSG`, and the lower
    window is a `magit-diff: $DIRNAME` window.

To rule out an issue with my `.emacs` file, I ran `magit-emacs-Q-command`, which produced the following:

````
/usr/bin/emacs -Q --eval \(setq\ debug-on-error\ t\) -L /home/nr/.emacs.d/elpa/dash-20210609.1330/ -L /home/nr/.emacs.d/elpa/transient-0.3.6/ -L /home/nr/.emacs.d/elpa/with-editor-20210524.1654/ -L /home/nr/.emacs.d/elpa/magit-3.1.1/ -L /home/nr/.emacs.d/elpa/magit-section-3.1.1/ -L /home/nr/.emacs.d/elpa/git-commit-20210525.844/ -l /home/nr/.emacs.d/elpa/magit-3.1.1/magit
````

Starting Emacs with this command confirms that Magit is opening `COMMIT_EDITMSG` in a new frame.

Unfortunately, @tarsius is not able to duplicate the issue.
Does anybody have ideas about my next step?

**Comment by npostavs** [comment]
Maybe `M-x debug-on-entry make-frame RET` and post the backtrace?

**Reply by nrnrnr** [reply]
Nice idea!  It looks like something---I don't know what---might be invoking `emacsclient`.

The backtrace is

```
Debugger entered--entering a function:
* make-frame(((display . ":0") (client . #<process server <1>>) (environment "A__z=\"*SHLVL" "VISUAL=emacs" "USER=nr" "TERM=dumb" "SHLVL=1" "PWD=/home/nr/src/lua" "PATH=/usr/lib/git-core:/home/nr/.opam/default/bin:..." "INSIDE_EMACS=27.1,magit" "HOME=/home/nr" "GIT_PREFIX=" "GIT_PAGER=cat" "GIT_LITERAL_PATHSPECS=1" "GIT_INDEX_FILE=.git/index" "GIT_EXEC_PATH=/usr/lib/git-core" "GIT_EDITOR=/home/nr/bin/emacsclient --socket-name=..." "GIT_CONFIG_PARAMETERS='core.preloadindex=true' 'lo..." "GIT_AUTHOR_NAME=Norman Ramsey" "GIT_AUTHOR_EMAIL=nr@cs.tufts.edu" "GIT_AUTHOR_DATE=@1626703599 -0400" "EDITOR=emacs" "DISPLAY=:0" "ALTERNATE_EDITOR=sh -c 'printf \"WITH-EDITOR: $$ OP..." "_=*24420*/usr/bin/emacsclient")))
  make-frame-on-display(":0" ((client . #<process server <1>>) (environment "A__z=\"*SHLVL" "VISUAL=emacs" "USER=nr" "TERM=dumb" "SHLVL=1" "PWD=/home/nr/src/lua" "PATH=/usr/lib/git-core:/home/nr/.opam/default/bin:..." "INSIDE_EMACS=27.1,magit" "HOME=/home/nr" "GIT_PREFIX=" "GIT_PAGER=cat" "GIT_LITERAL_PATHSPECS=1" "GIT_INDEX_FILE=.git/index" "GIT_EXEC_PATH=/usr/lib/git-core" "GIT_EDITOR=/home/nr/bin/emacsclient --socket-name=..." "GIT_CONFIG_PARAMETERS='core.preloadindex=true' 'lo..." "GIT_AUTHOR_NAME=Norman Ramsey" "GIT_AUTHOR_EMAIL=nr@cs.tufts.edu" "GIT_AUTHOR_DATE=@1626703599 -0400" "EDITOR=emacs" "DISPLAY=:0" "ALTERNATE_EDITOR=sh -c 'printf \"WITH-EDITOR: $$ OP..." "_=*24420*/usr/bin/emacsclient")))
  server-create-window-system-frame(":0" nil #<process server <1>> nil nil)
  server-process-filter(#<process server <1>> "-env _=*24420*/usr/bin/emacsclient -env ALTERNATE_...")
```

The command that invokes Emacs is

```
env -i EDITOR=emacs TERM="$TERM" DISPLAY="$DISPLAY" PATH="$PATH" USER="$USER" HOME="$HOME" VISUAL=emacs /usr/bin/emacs -Q --eval \(setq\ debug-on-error\ t\) -L /home/nr/.emacs.d/elpa/dash-20210609.1330/ -L /home/nr/.emacs.d/elpa/transient-0.3.6/ -L /home/nr/.emacs.d/elpa/with-editor-20210524.1654/ -L /home/nr/.emacs.d/elpa/magit-3.1.1/ -L /home/nr/.emacs.d/elpa/magit-section-3.1.1/ -L /home/nr/.emacs.d/elpa/git-commit-20210525.844/ -l /home/nr/.emacs.d/elpa/magit-3.1.1/magit ~/src/lua/
```

**Reply by nrnrnr** [reply]
I poked around a bit.  It looks like `emacsclient` is being invoked by something in the `with-editor` package.  There's an option to `server-process-filter` that would suppress the creation of a new frame, but poking at `with-editor.el`, I don't quite see where the `emacsclient` command is being put together.  My elisp skills are limited, and the waters are getting a little deep there.

**Reply by tarsius** [reply]
Please post the output of `M-x with-editor-debug RET`. How to you install Emacs?

**Comment by nrnrnr** [comment, marked as answer]
**SOLVED**.  I have a wrapper around `emacsclient` that's using the `-c` option.
